src/plugins/data/public/search/fetch/types.ts

```diff
@@ -20,9 +20,6 @@
 import { IUiSettingsClient } from '../../../../../core/public';
 import { ISearchStartLegacy } from '../types';
 
-export type SearchRequest = any;
```

Comment:
Also I think we could make this somewhat stricter than `Record<string, any>` because those record values must be something that's serializable.

This is how we've handled similar situations elsewhere: 
https://github.com/elastic/kibana/blob/a987cef5835ef2b94375becdb47b43d63c5d811e/src/plugins/expressions/common/expression_types/specs/datatable.ts#L26-L31

So you could do something like `type SearchRequest = SerializableState;`

In fact, we already have `SerializableState` defined for aggs in `data/common/search/aggs/agg_config.ts` as well, so it could possibly be located somewhere shared like `data/common/search/types.ts` and used in both places.

Reply:
I made a similar change (not exactly the same, because this type is actually **not serializable**.
Added a comment to clarify.